rmgpy/molecule/group.py

```diff
@@ -1262,11 +1273,42 @@ def specifyAtomExtensions(self,i,basename,R):
             if len(old_atom_type ) > 1:
                 old_atom_type_str = ''
                 for k in old_atom_type:
-                    old_atom_type_str += k.label
+                    labelList.append(k.label)
+                for p in sorted(labelList):
+                    old_atom_type_str += p
             else:
                 old_atom_type_str = old_atom_type[0].label
 
-            grps.append((grp,grpc,basename+'_'+old_atom_type_str+'->'+item.label,'atomExt',(i,)))
+            grps.append((grp,grpc,basename+'_'+str(i+1)+old_atom_type_str+'->'+item.label,'atomExt',(i,)))
```

Comment:
Would it make sense to use `>` instead of `->` to save a character?

Reply:
Honestly, in practice the names are so long I've tried to err on the side of making it easy to read individual sections of the name (having a bit more detail) rather than the thing as a whole.